### ISSUE TITLE
Tween.animate() throws exception for Animation<Offset> in slow mode

### ISSUE THREAD

**Issue by adchang** [flutter/flutter#13812]
## Steps to Reproduce

Use a FractionalOffsetTween.animate() for position parameter in SlideTransition

```dart
new SlideTransition(
  position: new FractionalOffsetTween(
    begin: FractionalOffset.topRight,
    end: FractionalOffset.topLeft,
  ).animate(
    new CurvedAnimation(
      parent: animation,
      curve: Curves.ease,
    )
  ),
...
)
```

## Logs

```
I/flutter (14009): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter (14009): The following assertion was thrown building
I/flutter (14009): _ModalScope-[LabeledGlobalKey<_ModalScopeState>#472a3](dirty, state: _ModalScopeState#d602a):
I/flutter (14009): type '_AnimatedEvaluation<FractionalOffset>' is not a subtype of type 'Animation<Offset>' of
I/flutter (14009): 'position' where
I/flutter (14009):   _AnimatedEvaluation is from package:flutter/src/animation/tween.dart
I/flutter (14009):   FractionalOffset is from package:flutter/src/painting/fractional_offset.dart
I/flutter (14009):   double is from dart:core
I/flutter (14009):   FractionalOffset is from package:flutter/src/painting/fractional_offset.dart
I/flutter (14009):   Animation is from package:flutter/src/animation/animation.dart
I/flutter (14009):   Offset is from dart:ui
I/flutter (14009): 
I/flutter (14009): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (14009): more information in this error message to help you determine and fix the underlying cause.
I/flutter (14009): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (14009):   https://github.com/flutter/flutter/issues/new
I/flutter (14009): 
I/flutter (14009): When the exception was thrown, this was the stack:
I/flutter (14009): #0      new SlideTransition (package:flutter/src/widgets/transitions.dart:116)
I/flutter (14009): #1      RTLPageRoute.buildTransitions (package:codexinth_honeybee/src/flutter/page_route/rtl_page_route.dart:29)
I/flutter (14009): #2      _ModalScopeState.build (package:flutter/src/widgets/routes.dart:469)
I/flutter (14009): #3      StatefulElement.build (package:flutter/src/widgets/framework.dart:3634)
I/flutter (14009): #4      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3544)
I/flutter (14009): #5      Element.rebuild (package:flutter/src/widgets/framework.dart:3445)
I/flutter (14009): #6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3524)
I/flutter (14009): #7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3662)
I/flutter (14009): #8      ComponentElement.mount (package:flutter/src/widgets/framework.dart:3519)
I/flutter (14009): #9      Element.inflateWidget (package:flutter/src/widgets/framework.dart:2857)
I/flutter (14009): #10     Element.updateChild (package:flutter/src/widgets/framework.dart:2660)
I/flutter (14009): #11     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3556)
I/flutter (14009): #12     Element.rebuild (package:flutter/src/widgets/framework.dart:3445)
I/flutter (14009): #13     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3524)
I/flutter (14009): #14     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3662)
I/flutter (14009): #15     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3519)
I/flutter (14009): #16     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2857)
I/flutter (14009): #17     Element.updateChild (package:flutter/src/widgets/framework.dart:2660)
I/flutter (14009): #18     RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:4319)
I/flutter (14009): #19     MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4647)
I/flutter (14009): #20     Element.updateChild (package:flutter/src/widgets/framework.dart:2649)
I/flutter (14009): #21     _TheatreElement.update (package:flutter/src/widgets/overlay.dart:507)
I/flutter (14009): #22     Element.updateChild (package:flutter/src/widgets/framework.dart:2649)
I/flutter (14009): #23     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3556)
I/flutter (14009): #24     Element.rebuild (package:flutter/src/widgets/framework.dart:3445)
I/flutter (14009): #25     StatefulElement.update (package:flutter/src/widgets/framework.dart:3681)
I/flutter (14009): #26     Element.updateChild (package:flutter/src/widgets/framework.dart:2649)
I/flutter (14009): #27     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3556)
I/flutter (14009): #28     Element.rebuild (package:flutter/src/widgets/framework.dart:3445)
I/flutter (14009): #29     ProxyElement.update (package:flutter/src/widgets/framework.dart:3791)
I/flutter (14009): #30     Element.updateChild (package:flutter/src/widgets/framework.dart:2649)
I/flutter (14009): #31     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4539)
I/flutter (14009): #32     Element.updateChild (package:flutter/src/widgets/framework.dart:2649)
I/flutter (14009): #33     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3556)
I/flutter (14009): #34     Element.rebuild (package:flutter/src/widgets/framework.dart:3445)
I/flutter (14009): #35     StatefulElement.update (package:flutter/src/widgets/framework.dart:3681)
I/flutter (14009): #36     Element.updateChild (package:flutter/src/widgets/framework.dart:2649)
I/flutter (14009): #37     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4539)
I/flutter (14009): #38     Element.updateChild (package:flutter/src/widgets/framework.dart:2649)
I/flutter (14009): #39     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:4539)
I/flutter (14009): #40     Element.updateChild (package:flutter/src/widgets/framework.dart:2649)
I/flutter (14009): #41     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3556)
I/flutter (14009): #42     Element.rebuild (package:flutter/src/widgets/framework.dart:3445)
I/flutter (14009): #43     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2193)
I/flutter (14009): #44     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:611)
I/flutter (14009): #45     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:203)
I/flutter (14009): #46     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:916)
I/flutter (14009): #47     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:856)
I/flutter (14009): #48     BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:768)
I/flutter (14009): #49     _invoke (file:///b/build/slave/Linux_Engine/build/src/flutter/lib/ui/hooks.dart:113)
I/flutter (14009): #50     _drawFrame (file:///b/build/slave/Linux_Engine/build/src/flutter/lib/ui/hooks.dart:102)
```

## Flutter Doctor

adchang@bacchus:mobile$ flutter doctor
[✓] Flutter (on Linux, locale en_US.UTF-8, channel alpha)
    • Flutter at /opt/adchang/sdk/flutter
    • Framework revision 8f65fec5f5 (3 weeks ago), 2017-12-12 09:50:14 -0800
    • Engine revision edaecdc8b8
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.d8ae797298c3a6cf8dc9f4558707bd2672224d3e

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.3)
    • Android SDK at /opt/adchang/sdk/android
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-26, build-tools 26.0.3
    • ANDROID_HOME = /opt/adchang/sdk/android
    • Java binary at: /opt/adchang/bin/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[✓] Android Studio (version 3.0)
    • Android Studio at /opt/adchang/bin/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[-] Connected devices
    • None

**Comment by Hixie** [comment]
SlideTransition was changed recently to take an `Animation<Offset>` rather than an `Animation<FractionalOffset>`.

**Comment by Hixie** [comment]
See https://groups.google.com/forum/#!topic/flutter-dev/fmr-C9xK5t4

**Comment by adchang** [comment]
Thanks Hixie!

For those interested in a RTL transition, replace the FractionalOffsetTween with:

new Tween<Offset>(
  begin: const Offset(1.0, 0.0),
  end: Offset.zero,
)

Additional commentary:
I actually did see the post regarding the introduction of Alignment vs FractionalOffset when I started seeing the exception in slow mode (I didn't catch it earlier as I've been using release mode, which works fine). DO NOT simply replace FractionalOffsetTween with https://docs.flutter.io/flutter/rendering/AlignmentTween-class.html, which as the documentation says, creates a fractional offset tween, and as a result, you will see the same exception.
